lib/sync_checker/formats/html_attachment_check.rb

```diff
@@ -98,8 +106,30 @@ def top_level_fields_hash
         }
       end
 
+      def attachment_should_exist_in_live_content_store?
+        Edition::PUBLICLY_VISIBLE_STATES.include?(attachable.state) ||
+          attachable_is_unpublished?
```

Comment:
Yes in terms of the actual content item but if they are unpublished at all we still want to run the `HtmlAttachmentUnpublishedCheck` against the live content store to check for the gone or redirect. The other checks for live will ignore it if content store returns one of those.